drv/transceivers-server/src/udp.rs

```diff
@@ -450,9 +450,11 @@ impl ServerImpl {
             .transceivers
             .wait_and_check_i2c(mask)
             .map_err(|_e| HwError::WaitFailed)?;
-        if !err_mask.left.is_empty() || !err_mask.right.is_empty() {
+        if !err_mask.left.is_empty() {
             // FPGA reported an I2C error
-            return Err(HwError::I2cError);
+            return Err(HwError::I2cError(PortMask(err_mask.left.get())));
+        } else if !err_mask.right.is_empty() {
+            return Err(HwError::I2cError(PortMask(err_mask.right.get())));
```

Comment:
How can the host tell if the error occurred in the left or right FPGA?

Reply:
I _think_ the host can only talk to one FPGA at a time (`-f 0` or `-f 1`) anyway, so it will know which FPGA the error mask is for because it will be whichever one it tried to talk to.